### PR TITLE
fix issue 15631 - gdb: Parent's scope not considered for symbol lookup

### PR DESCRIPTION
**Pull request by rainers**
add basic inheritance information to class type dwarf debug info

Alternative to https://github.com/dlang/dmd/pull/12925

### CODE REVIEW

**Comment by UplinkCoder** [comment]
I like how this looks :+1:

**Review by UplinkCoder — APPROVED** [review]
Looks like a good interim solution.

**Comment by UplinkCoder** [comment]
@Geod24 you will like this.

**Review by RazvanN7 — APPROVED** [review]

**Comment by UplinkCoder** [comment]
@ibuclaw  @RazvanN7 seems like the CirrusCI FreeBSD has not reported the state.

**Comment by ibuclaw** [comment]
You need to rebase against master to pick up the new freebsd image.

File: test/runnable/gdb_baseclass_fields.d [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+REQUIRED_ARGS: -g
+PERMUTE_ARGS:
+GDB_SCRIPT:
+---
+b 30
```
Comment (ibuclaw) [inline test/runnable/gdb_baseclass_fields.d:6]: Hmm, this doesn't work on gdb 12, see #12499.

**Comment by ibuclaw** [comment]
This PR caused a testsuite regression of https://issues.dlang.org/show_bug.cgi?id=21874

**Pull request merged**